packages/svelte/src/compiler/phases/3-transform/client/visitors/shared/utils.js

```diff
@@ -394,3 +394,36 @@ export function build_expression(context, expression, metadata, state = context.
 
 	return sequence;
 }
+
+/**
+ * Wraps a statement/expression with dev stack tracking in dev mode
+ * @param {CallExpression} call_expression - The function call to wrap (e.g., $.if, $.each, etc.)
+ * @param {{ start?: number }} node - AST node for location info
+ * @param {'component' | 'if' | 'each' | 'await' | 'key'} type - Type of block/component
+ * @param {Record<string, number | string>} [additional] - Any additional properties to add to the dev stack entry
+ * @returns {Statement} - Statement with or without dev stack wrapping
+ */
+export function with_dev_stack(call_expression, node, type, additional) {
+	if (!dev) {
+		return b.stmt(call_expression);
+	}
+
+	const location = node.start && locator(node.start);
+	if (!location) {
+		return b.stmt(call_expression);
+	}
+
+	return b.stmt(
+		b.call(
+			'$.with_dev_stack',
+			b.arrow([], b.block([b.stmt(call_expression)])),
+			b.literal(type),
+			b.literal(filename),
```

Comment:
it'd be nice to use `ComponentName` here rather than printing the literal filename, though as things stand that would involve passing the `analysis` object into this function. I think it would be good if `analysis` was globally available in the same way that things like `dev` and `locator` are, though that would mean shuffling some stuff around... might see how straightforward that is in a separate PR

Reply:
Component name as in...? Do you mean `<Foo ..>` -> `Foo`, `<x.y ...` -> `x.y`?

Reply:
I mean that you pass in `Foo`, and inside `with_dev_stack` we do `Component[FILENAME]`, rather than passing the actual filename around

Reply:
Oh I see